pkg/apis/scaler/v1alpha2/types_demand.go

```diff
@@ -88,6 +91,9 @@ type DemandSpec struct {
 	// EnforceSingleZoneScheduling indicates this demand must
 	// be satisfied in a single zone.
 	EnforceSingleZoneScheduling bool `json:"enforce-single-zone-scheduling"`
+	// Zone indicates the zone where the Demand should be fulfilled.
+	// If this parameter is not set, the Demand can be fulfilled in any zone.
+	Zone *string `json:"zone,omitempty"`
```

Comment:
should this be `string`?

Reply:
Changed this to Zone for now (no pointer).

I believe this may have been to check optionality by checking for a nil pointer instead of the empty string, will let @sguilhemducleon chime in though!

Reply:
Will you always set this field? Missing properties are usually represented by pointers instead of empty strings, see https://dhdersch.github.io/golang/2016/01/23/golang-when-to-use-string-pointers.html.

Here in both cases, the property will be marshalled as a missing property from the json (omitempty), so the difference is only during unmarshalling whether we want an empty string or a nil when the json property is missing.

I would be in favor of using *Zone instead, I think it's more explicit that we need to check for nils, empty strings are error-prone.

Reply:
👍 The vast majority of the time, this will not be set. I'll make it *Zone